karoo_gp/base_class.py

```diff
@@ -182,9 +182,11 @@ def log(self, msg, display={'i', 'g', 'm', 'db'}):
     def pause(self, display={'i', 'g', 'm', 'db'}):
         if not self.pause_callback:
             self.log('No pause callback function provided')
-            return
-        if self.display in display or display is None:
+            return 0
+        elif self.display in display or display is None:
             self.pause_callback(self)
+        else:
+            return 0
```

Comment:
Why `0`?

Reply:
The loop in `gp.fit()` starts with `while menu != 0`.

If `gp.mode != 's'` (server), `gp.pause()` is called after `gen_max` loops. If also `gp.display == 's'` (silent), we don't want the pause menu to show, but it still needs to exit the loop.